server/server/src/sync/sync_api_v5.rs

```diff
@@ -97,13 +98,13 @@ impl SyncApiV5 {
         server_url: Url,
         credentials: SyncCredentials,
         client: Client,
-        hardware_id: &str,
+        hardware_id: AppData,
```

Comment:
The &str argument is better here. This sync api file is just a wrapper for the api but is otherwise remote server independent, i.e. it shouldn't have other app dependencies. For example, you could take this file (and the api relate SyncCredntials) and use it somewhere else.